UPGRADE.md

```diff
@@ -2666,6 +2666,23 @@ const ReviewEditToolbar = (props: ToolbarProps<Review>) => {
 };
 ```
 
+Since the `<Toolbar>` component is not cloned anymore by react-admin layouts, properties like `mutationMode` and `submitOnEnter` are not passed down from parent components. So, if you rely on this behaviour, you should create your own custom toolbar instead
+
+```diff
+import { Toolbar, SimpleForm, Edit, TextInput } from 'react-admin';
+
++const MyToolbar = props => <Toolbar {...props} mutationMode="pessimistic" submitOnEnter={false} />;
```

Comment:
Actually, I think the Toolbar shouldn't clone and inject props either

Reply:
So, the solution should be to pass a custom SaveButton/DeleteButton inside a custom Toolbar?